src/workerd/io/worker.c++

```diff
@@ -2605,18 +2606,31 @@ struct Worker::Actor::Impl final: public kj::TaskSet::ErrorHandler {
   kj::Maybe<RunningAlarm> runningAlarm;
   // Used to handle deduplication of alarm requests
 
-  Impl(Worker::Actor& self, Worker::Lock& lock, Actor::Id actorId,
+  Impl(const Worker& workerRef, LockType lockType, Actor::Id actorId,
```

Comment:
I think you should still just take the `Worker::Lock&`. You can get the `Worker&` from it via `getWorker()`.

Reply:
Addressed both comments in one commit https://github.com/cloudflare/workerd/pull/125/commits/7044c0d9ca63e54bf800a4481acbc619a581389b